resolver/error.go

```diff
@@ -22,13 +22,52 @@ func (e *Error) Error() string {
 	return fmt.Sprintf("grpc-federation: %s", strings.Join(errs, ": "))
 }
 
+func (e *Error) add(err error) {
+	if err == nil {
+		return
+	}
+
+	e.Errs = append(e.Errs, err)
+}
+
 // LocationError holds error message with location.
-type LocationError struct {
+type LocationError interface {
+	error
+	GetLocation() *source.Location
+	GetMessage() string
+}
+
+type LocationErrorImpl struct {
+	Location *source.Location
+	Message  string
+}
+
+func (e *LocationErrorImpl) GetLocation() *source.Location {
+	return e.Location
+}
+
+func (e *LocationErrorImpl) GetMessage() string {
+	return e.Message
+}
+
+func (e *LocationErrorImpl) Error() string {
+	return e.Message
+}
+
+type CyclicDependencyError struct {
```

Comment:
We may be able to contain more valuable information for users in case of cyclic dependency such as dependency path.